app/Http/Controllers/StatController.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Click;
+use App\Models\Link;
+use Illuminate\Http\RedirectResponse;
+use Illuminate\Support\Facades\DB;
+use Illuminate\View\View;
+
+class StatController extends Controller
+{
+    /**
+     * @param string $linkId
+     * @return RedirectResponse|View
+     */
+    public function listClicks($linkId)
+    {
+        if (!$link = Link::find($linkId))
+        {
+            return redirect(route('admin'))->with('error', "Such link does not exist.");
+        }
+
+        if ($check = $this->performAuthCheck($link))
+        {
+            return $check;
+        }
+
+        $clicks = Click::where('link_id', $link->id)->orderBy('date', 'desc')->paginate(15);
+
+        return view('clicks', ['link' => $link, 'clicks' => $clicks]);
+    }
+
+    /**
+     * @param string $type
+     * @param string $linkId
+     * @return RedirectResponse|View
+     */
+    public function makeReport($type, $linkId)
+    {
+        if (!in_array($type,['day', 'country', 'referer']))
+        {
+            return redirect(route('admin'))->with('error', "Such report does not exist.");
+        }
+
+        if (!$link = Link::find($linkId))
+        {
+            return redirect(route('admin'))->with('error', "Such link does not exist.");
+        }
+
+        if ($check = $this->performAuthCheck($link))
+        {
+            return $check;
+        }
+
+        switch ($type) {
+            case 'day':
+                $stats = DB::table('clicks')
+                    ->select(DB::raw("DATE_FORMAT(date, '%Y-%m-%d') as date, count(*) as clicks"))
```

Comment:
It would be preferable to avoid using raw SQL to perform these queries. Is it possible to use Eloquent's built in functions to retrieve this information rather than performing raw queries? https://laravel.com/docs/5.1/eloquent